tests/test_needtable.py

```diff
@@ -16,27 +16,28 @@ def test_doc_build_html(app, status, warning):
     assert "Test table caption" in html
 
 
-# ToDo: Make this not so html specific, as classes and order of attributes may change
-#     # check table caption and table wrapped into figure
-#     if sphinx.version_info[0] >= 4:
-#         figure = """
-# <figcaption>
-# <p><span class="caption-text">Test table caption
-# """
-#     else:
-#         figure = """
-# <figure id="needtable-index-0">
-# <figcaption>
-# <p><span class="caption-text">Test table caption
-# """
-#     assert figure in html
-#
-#     # negative test to check table without caption
-#     if sphinx.version_info[0] >= 4:
-#         assert '<table class="NEEDS_TABLE rtd-exclude-wy-table docutils align-default" id="needtable-index-1">'
-#         in html
-#     else:
-#         assert '<table class="NEEDS_TABLE rtd-exclude-wy-table docutils" id="needtable-index-1">' in html
+@with_app(buildername="xml", srcdir="doc_test/doc_needtable")
```

Comment:
you can/should keep html here.
But use ElementTree to analyse it.

Reply:
Or maybe leave it in the existing test-case (as it was before), but us elementree-checks instead of string-compares.

Reply:
Done.